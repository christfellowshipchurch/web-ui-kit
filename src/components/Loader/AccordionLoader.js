import React from 'react'

import {
  accordionLoaderContainer, accordionLoaderColumn, accordionLoaderItem
} from '../css/styles.module.css'

const AccordionLoader = () => (
  <div className={accordionLoaderContainer}>
    {[1, 2, 3, 4, 5, 6].map((n) => (
      <div className={accordionLoaderColumn} key={n}>
        <div className={accordionLoaderItem}></div>
      </div>
    ))}
  </div>
)

const propTypes = {}
const defaultProps = {}


AccordionLoader.propTypes = propTypes
AccordionLoader.defaultProps = defaultProps

export default AccordionLoader
