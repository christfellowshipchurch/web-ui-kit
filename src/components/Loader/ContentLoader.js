import React from 'react'

import {
  contentLoaderContainer, contentLoaderImg, contentLoaderTitle, contentLoaderLine,
} from '../css/styles.module.css'

const ContentLoader = () => (
  <div className={contentLoaderContainer}>

    <div className={contentLoaderImg}></div>

    <div className={contentLoaderTitle}></div>

    <div className={contentLoaderLine}></div>
    <div className={contentLoaderLine}></div>
    <div className={contentLoaderLine}></div>
    <div className={contentLoaderLine}></div>
    <div className={contentLoaderLine}></div>
  </div>
)

const defaultProps = {}

const propTypes = {}

ContentLoader.defaultProps = defaultProps
ContentLoader.propTypes = propTypes

export default ContentLoader;
