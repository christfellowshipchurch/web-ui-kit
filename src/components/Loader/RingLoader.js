import React from 'react'
import {
    ringContainer, ringLoader
} from '../styles/scss/styles.module.scss'

const RingLoader = () => <div className={ringContainer}><div className={ringLoader}></div></div>

const propTypes = {}
const defaultProps = {}

RingLoader.propTypes = propTypes
RingLoader.defaultProps = defaultProps

export default RingLoader
