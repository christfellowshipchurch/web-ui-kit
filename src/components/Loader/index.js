import Ring from './RingLoader'
import ContentLoader from './ContentLoader'
import AccordionLoader from './AccordionLoader'

const Loaders = Ring
Loaders.Content = ContentLoader
Loaders.Accordion = AccordionLoader

export default Loaders