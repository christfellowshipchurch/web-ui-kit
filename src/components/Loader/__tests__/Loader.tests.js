import React from 'react'
import Loader from '../../Loader'
import renderer from 'react-test-renderer'

describe('Loader', () => {
    it('renders the ring loader as the default loader', () => {
        const tree = renderer.create(<Loader />)

        expect(tree).toMatchSnapshot()
    })

    it('renders an accordion styled loader', () => {
        const tree = renderer.create(<Loader.Accordion />)

        expect(tree).toMatchSnapshot()
    })

    it('renders a content styled loader', () => {
        const tree = renderer.create(<Loader.Content />)

        expect(tree).toMatchSnapshot()
    })
})