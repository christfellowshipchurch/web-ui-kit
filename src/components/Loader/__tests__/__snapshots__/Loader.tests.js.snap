// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders a content styled loader 1`] = `
<div>
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
  <div />
</div>
`;

exports[`Loader renders an accordion styled loader 1`] = `
<div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
  <div>
    <div />
  </div>
</div>
`;

exports[`Loader renders the ring loader as the default loader 1`] = `
<div>
  <div />
</div>
`;
